src/scm/github.js

```diff
@@ -0,0 +1,433 @@
+        }));
+        return {
+          tree: tree
+        };
+      })
+      .then((payload) => {
+        return $.ajax({
+          url: `${this.baseUrl}/repos/${context.repo.fullName}/git/trees`,
+          headers: {
+            'Authorization': `token ${this.accessToken}`
+          },
+          method: 'POST',
+          crossDomain: true,
+          dataType: 'json',
+          contentType: 'application/json',
+          data: JSON.stringify(payload)
+        });
+      })
+      .then((response) => {
+        return Object.assign(response, { parent: responses[1].commit.sha })
+      })
+      .fail((err) => {
+        throw err;
+      })
+    })
+    .then((response) => {
+      const payload = {
+        message: $('#commit-comment').val(),
+        tree: response.sha,
+        parents: [
+          response.parent
+        ]
+      };
+      return $.ajax({
+        url: `${this.baseUrl}/repos/${context.repo.fullName}/git/commits`,
+        headers: {
+          'Authorization': `token ${this.accessToken}`
+        },
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      });
+    })
+    .then((response) => {
+      const payload = {
+        force: true,
+        sha: response.sha
+      };
+      return $.ajax({
+        url: `${this.baseUrl}/repos/${context.repo.fullName}/git/refs/heads/${context.branch}`,
+        headers: {
+          'Authorization': `token ${this.accessToken}`
+        },
+        method: 'PATCH',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      });
+    })
+    .then(() => {
+      showAlert(`Successfully push to ${context.branch} of ${context.repo.name}`);
+    })
+    .catch((err) => {
+      showAlert('Failed to push', LEVEL_ERROR);
+    });
+  }
+
+  pushToGist(code) {
+    const files = $('.diff-file:checked').toArray().map((elem) => elem.value);
+    if (files.length === 0) {
+      showAlert('Nothing to do', LEVEL_WARN);
+      return;
+    }
+    const payload = {
+      files: {}
+    };
+    files.forEach((file) => {
+      payload.files[file] = {
+        content: code.gas[file]
+      };
+    })
+    if (code.github['init_by_gas_hub.html']) {
+      payload.files['init_by_gas_hub.html'] = null;
+    }
+    if ($('#gist-desc').val() !== '') {
+      payload.description = $('#gist-desc').val();
+    }
+    console.log(payload);
+    return $.ajax({
+      url: `${this.baseUrl}/gists/${context.branch}`,
+      headers: {
+        'Authorization': `token ${this.accessToken}`
+      },
+      method: 'PATCH',
+      crossDomain: true,
+      dataType: 'json',
+      contentType: 'application/json',
+      data: JSON.stringify(payload)
+    })
+    .then(() => {
+      showAlert(`Successfully update gist: ${context.branch}`);
+    })
+    .fail((err) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/65105829498a48107e9a8b01cb705b310fffdc68/src/scm/github.js#L174"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>